src/main/scala/nl.knaw.dans.easy.solr4files/ApplicationWiring.scala

```diff
@@ -15,15 +15,79 @@
  */
 package nl.knaw.dans.easy.solr4files
 
+import java.net.{URI, URL}
+
+import nl.knaw.dans.easy.solr4files.components._
+import nl.knaw.dans.lib.error.{CompositeException, TraversableTryExtensions}
 import nl.knaw.dans.lib.logging.DebugEnhancedLogging
 
+import scala.util.Try
+import scala.xml.Elem
+
 /**
  * Initializes and wires together the components of this application.
  *
  * @param configuration the application configuration
  */
-class ApplicationWiring(configuration: Configuration) extends DebugEnhancedLogging
-// Mix in components
-{
-  // Configure components with configuration settings.
+class ApplicationWiring(configuration: Configuration)
+  extends DebugEnhancedLogging with Vault with Solr {
+
+  // don't need resolve for solr, URL gives more early errors TODO perhaps not enough early errors
+  override val solrUrl: URL = new URL(configuration.properties.getString("solr.url", ""))
+  override val vaultBaseUri: URI = new URI(configuration.properties.getString("vault.url", ""))
+
+  def initAllStores(): Try[FeedBackMessage] = {
+    getStoreNames
+      .flatMap(updateStores)
+      .map(results => s"Updated all bags of ${ results.size } stores ($vaultBaseUri)")
+  }
+
+  def initSingleStore(storeName: String): Try[FeedBackMessage] = {
+    getBagIds(storeName)
+      .flatMap(updateBags(storeName, _))
+      .map(results => s"Updated ${ results.size } bags of one store ($storeName)")
+  }
+
+  def update(storeName: String, bagId: String): Try[FeedBackMessage] = {
+    val bag = Bag(storeName, bagId, this)
+    for {
+      ddmXML <- bag.loadDDM
+      ddm = new DDM(ddmXML)
+      filesXML <- bag.loadFilesXML
+      _ <- deleteBag(bag.bagId)
+      feedbackMessage <- updateFiles(bag, ddm, filesXML)
+      _ <- commit()
+    } yield feedbackMessage
+  }
+
+  def delete(bagId: String): Try[FeedBackMessage] = for {
+    _ <- deleteBag(bagId)
+    _ <- commit()
```

Comment:
What happens if some of the files have been deleted, but there is an exception and we do not do the commit. Then those deletes are not committed, but will that be done with the next command if we run as a service..? Maybe we should try a commit if something fails, or we need to understand what happens if we do not commit.